sotrplib/sources/sources.py

```diff
@@ -1,4 +1,252 @@
-from pydantic import BaseModel
+from typing import Any, Literal, Optional
+
+import structlog
+from astropy import units as u
+from astropydantic import AstroPydanticQuantity
+from pydantic import BaseModel, Field
+from structlog.types import FilteringBoundLogger
+
+
+class BaseSource(BaseModel):
+    ra: AstroPydanticQuantity[u.deg]
+    dec: AstroPydanticQuantity[u.deg]
+    flux: Optional[AstroPydanticQuantity[u.mJy]] = None
+
+
+class RegisteredSource(BaseSource):
+    """
+    A registered source object which may have catalog crossmatches.
+    This object has a source_id which may belong to the catalog and a source type.
+
+    This object need not have a flux.
+
+    Extendedness and positional uncertainty is stored if available.
+    """
+
+    source_id: str | None = None
+    source_type: Literal["Extragalactic", "Star", "Asteroid", "Unknown"] | None = None
+
+    crossmatch_names: list | None = None
+    crossmatch_probabilities: list | None = None
+    crossmatch_fluxes: list | None = None
+    crossmatch_frequencies: list | None = None
+
+    extended: bool | None = None
+    err_ra: AstroPydanticQuantity[u.deg] | None = None
+    err_dec: AstroPydanticQuantity[u.deg] | None = None
+    log: FilteringBoundLogger | None = Field(default_factory=structlog.get_logger)
+
+    model_config = {"arbitrary_types_allowed": True}
+
+    def add_crossmatches(
+        self,
+        new_names: list,
+        new_probabilities: list = None,
+        new_fluxes: list = None,
+        new_frequencies: list = None,
```

Comment:
`new` is redundant in the variable names.

Reply:
Not sure this function works well. It seems that you're adding new crossmatches here to an already existing list. How are we to interpret those probabilities in this case?

I think you should probably just force the caller into generating their own new list of probabilities. Then that can be assigned to RegisteredSource.crossmatches.